tests/Unit/Carrier/Factory/CarrierFactoryTest.php

```diff
@@ -0,0 +1,370 @@
+                    'shipmentOptions' => [
+                        'labelDescription' => [
+                            'type'          => 'string',
+                            'minimumLength' => 0,
+                            'maximumLength' => 45,
+                        ],
+                    ],
+                ],
+            ],
+        ],
+        [
+            'id'               => 4,
+            'name'             => 'dpd',
+            'contractId'       => 10932621,
+            'primary'          => 0,
+            'type'             => 'custom',
+            'recipientOptions' => [
+                [
+                    'packageTypeId'   => 1,
+                    'packageTypeName' => 'package',
+                    'deliveryTypes'   => [
+                        [
+                            'id'   => 2,
+                            'name' => 'standard',
+                        ],
+                    ],
+                    'shipmentOptions' => [
+                        'insurance'        => [
+                            'type' => 'integer',
+                            'enum' => [
+                                520,
+                            ],
+                        ],
+                        'labelDescription' => [
+                            'type'          => 'string',
+                            'minimumLength' => 0,
+                            'maximumLength' => 45,
+                        ],
+                    ],
+                ],
+            ],
+            'returnOptions'    => [],
+        ],
+    ],
+];
+
+it('creates empty carrier and log', function () {
+    PdkFactory::create(MockConfig::DEFAULT_CONFIG);
+
+    $carrier = CarrierFactory::create('thisisnocarrier', MOCK_CONFIG);
+
+    expect($carrier->getName())
+        ->toBe(null)
+        ->and($carrier->getType())
+        ->toBe(null)
+        ->and(DefaultLogger::getLogs())
+        ->toBe([
+            [
+                'level'   => 'warning',
+                'message' => "[PDK]: Could not find any carrier inside config",
+                'context' => [
+                    'carrier' => 'thisisnocarrier',
+                    'config'  => MOCK_CONFIG,
+                ],
+            ],
+        ]);
+});
+
+it('creates main carrier by name', function () {
+    $carrier = CarrierFactory::create('postnl', MOCK_CONFIG);
+
+    expect($carrier->getName())
+        ->toBe('postnl')
+        ->and($carrier->getType())
+        ->toBe('main');
+});
+
+it('creates main carrier by carrierId', function () {
+    $carrier = CarrierFactory::create(5, MOCK_CONFIG);
+    expect($carrier->getName())
+        ->toBe('instabox')
+        ->and($carrier->getType())
+        ->toBe('main');
+});
+
+it('creates custom carrier by contractId', function () {
+    $carrier = CarrierFactory::create(10932621, MOCK_CONFIG);
+    expect($carrier->getName())
+        ->toBe('dpd')
+        ->and($carrier->getType())
+        ->toBe('custom');
+});
```

Comment:
nog even een test waarin je kijkt of de hele $carrier->toArray() wel klopt. gebruik hievoor Arr::dot() om een leesbaardere test te maken